src/index.ts

```diff
@@ -7,24 +7,103 @@ const points = new Array<{ x: number, y: number, segmentLength: number }>();
 const lines = new Array<{ x: number, y: number }>();
 let parameter = 0;
 let complexity = 0;
 let circles = false;
 let hasCapture = false;
 
+const parameterSlider = document.getElementById('parameter-slider') as HTMLInputElement;
+parameterSlider.oninput = function () {
+    parameter = parameterSlider.valueAsNumber;
+    redraw();
+};
+const complexityNumber = document.getElementById('complexity-number') as HTMLInputElement;
+complexityNumber.oninput = function () {
+    complexity = complexityNumber.valueAsNumber;
+    redraw();
+};
+const complexityCircles = document.getElementById('complexity-circles-check') as HTMLInputElement;
+complexityCircles.oninput = function () {
+    circles = complexityCircles.checked;
+    redraw();
+};
+
 function updateCanvasSize() {
     canvas.width = window.devicePixelRatio * canvas.clientWidth;
     canvas.height = window.devicePixelRatio * canvas.clientHeight;
 }
 
+function loadLocation() { //Inspiration from https://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript/21152762#21152762 (qd's not stored)
+    if (window.location.search)
+        window.location.search.substr(1).split(`&`)
+            .forEach(item => {
+                let [k, v] = item.split(`=`);
+                if (v != null) { //Restriction to valued keys
+                    v = v && decodeURIComponent(v);
+                    switch (true) {
+                        case (k === 'pt'):
+                            let [x, y] = v.split(';');
+                            if (x != null && y != null)
+                                addPoint(Number(x), Number(y), false);
+                            break;
+
+                        case (k === 'range'):
+                            parameterSlider.value = v;
+                            break;
+
+                        case (k === 'circles'):
+                            complexityCircles.checked = Boolean(Number(v));
+                            break;
+
+                        case (k === 'complexity'):
+                            complexityNumber.value = v;
+                            break;
+
+                        case (k === 'fftsize'):
+                            fftSize = Number(v);
+                            fft = new FFT(fftSize);
+                            input = fft.createComplexArray() as number[];
+                            output = fft.createComplexArray() as number[];
+                            break;
+
+                        default:
+                            break;
+                    }
+                }
+            })
+}
+
+function setLocation() {
+    if (points.length > 0 && 'URLSearchParams' in window) {
+        let pointsString: string = '';
+        points.forEach(pt => pointsString += '&pt=' + pt.x + ';' + pt.y);
+
+        var newRelativePathQuery = window.location.pathname + '?' + 'range=' + parameter + '&' + 'complexity=' + complexity + '&' + 'circles=' + Number(circles) + pointsString;
```

Comment:
Why not `'&complexity='` like in line 85?

Reply:
Ah, why is '&' not glued? So that I can change parameter's order without having to remember which one is the first one; `pointsString` is special, as I didn't special-cased the start/end point(s), nor wanted to get rid of an extra `&`. But yeah, `&complexity` would have been valid.
If that's a strong request, I may change that.

Reply:
Thanks, just curious. I'm fine with this.